bin/all_sky_search/pycbc_rerank_dq

```diff
@@ -0,0 +1,124 @@
+                     help="interferometer to analyze")
+parser.add_argument("--max-stat", type=float,default=8,
+                     help="maximum allowed dq value. "
+                           "Only used if --rate-file is not used. "
+                            "[default=8]")
+parser.add_argument("--input-file", required=True,nargs='+',
+                     help="List of files containing the dq timeseries")
+parser.add_argument("--rate-file", type = str, default = None,
+                     help="File containing the mapping from each "
+                           "dq bin to the trigger rate")
+parser.add_argument("--bank-file", type = str, default = None,
+                     help="File containing the template bank")
+parser.add_argument("--dq-type", type = str, default = 'dq',
+                     help="Name of dq type to be included in output file "
+                          "metadata [default='dq']")
+parser.add_argument("--output-file", required=True,
+                     help="name of output file")
+
+args = parser.parse_args()
+pycbc.init_logging(args.verbose)
+
+ifo = args.ifo
+
+log_like = numpy.array([])
+times = numpy.array([])
+for filename in args.input_file:
+    logging.info('Reading file %s...'%filename)
+    g = h5py.File(filename, 'r')
+    g_log_like = g[ifo + '/dq_val'][:]
+    g_times = g[ifo + '/times'][:]
+    g.close()
+    log_like = numpy.concatenate((log_like,g_log_like))
+    times = numpy.concatenate((times,g_times))
+
+log_like_ranking = numpy.argsort(log_like)
+len_log_like = len(log_like)
+
+times = times[log_like_ranking]
+log_like_vals = numpy.array(log_like[log_like_ranking])
+
+log_like_dict = {}
+percent_dict = {}
+
+if not args.rate_file or args.bank_file:
+    logging.error('Must supply either a bank or dq rates file')
+
+if args.rate_file is not None:
+    rates = {}
+    locs_dict = {}
+    if path.isfile(args.rate_file):
+        f = h5py.File(args.rate_file,'r')
+        bin_names = f.attrs['names'][:]
+        for bin_name in bin_names:
+            logging.info('Processing bin %s...'%bin_name)
+            rates = f['%s/rates'%bin_name][:]
+            locs_dict[bin_name] = f['%s/locs'%bin_name][:]
+            log_like_dict[bin_name] = numpy.zeros(len(log_like))
+
+            if not args.no_low_dq_cutoff:
+                rates = numpy.array([max(r,1.) for r in rates])
+
+            log_rates = numpy.log(rates)
+            percent_dict[bin_name] = log_rates
+
+            n_bins = len(log_rates)
+            percentiles = numpy.linspace(0,100,n_bins+1)
+            dq_percentiles = numpy.percentile(log_like_vals,percentiles)[1:]
+
+            rates_bin = numpy.array([n_bins - \
+                              len(dq_percentiles[dq_percentiles >= dq_ll]) \
+                              for dq_ll in log_like_vals])
+            log_like_dict[bin_name] = log_rates[rates_bin.astype('int')]
+
+        f.close()
+
+    else: logging.error('Could not find binned trigger rates file')
```

Comment:
Should this be an exception instead? Is it expected that the user (or workflow) might provide an invalid file path here?

Reply:
This has been made an exception